x-pack/plugin/esql/compute/ann/src/main/java/org/elasticsearch/compute/ann/MvCombiner.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.ann;
+
+/**
+ * When supporting multivalued fields, it is necessary to know how to combine them.
+ * This is done using a loop over the fields, calling combiner.combine(previous, value) on each, where the previous
+ * value was initialized to combiner.initial() before the loop.
+ * @param <T> the return type of the method that is annotated by the Evaluator annotation.
+ */
+public interface MvCombiner<T> {
+    T initial();
+
+    T combine(T previous, T value);
```

Comment:
If we invoke these over the interface there's a good chance the JVM'll run the invocation as megamorphic. The particular way you are doing it now runs into a edge case in my knowledge - because there's always a single subclass for each call site so, theoretically, it could know that this site always uses that subclass. I'm not sure if it does though.

Either way, I think it'd be better to declare the member variable as having the exact type of the implementation. So they'd be `AllCombiner combiner = new AllCombiner()` in the class. That way it *knows*.

I know it's premature optimization, but that's kind of the point of these `ExpressionEvaluator`s - to be as efficient as we can make them.

Also! I think the `<T>` is trouble here - at least at runtime. Maybe these should just be locked to `bool` for now.  I get that this is flexible so you can do a `BytesRefBuilder` or a `long` or whatever for combination, but for now we only need `bool` and I'm pretty sure the boxing/unboxing business gives optimizers trouble.

Reply:
This change has been implemented as a side effect of other changes I was making. We do now declare the evaluators field as the concrete type, not the interface, so hopefully these concerns are alleviated.

There might be some boxing/unboxing left, but hopefully that is a minor issue. I have taken care that we should only ever see that for multi-value cases, which right now is only for spatial search functions, so if there is any impact, it has a very narrow blast radius.